src/Tabs.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+import core from 'metal';
+import templates from './Tabs.soy';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+/**
+ * UI Component to navigate through tabbed data.
+ */
+class Tabs extends Component {
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {Object} tab
+	 * @param {number} index
+	 */
+	addTab(tab, index) {
+		if (core.isNumber(index)) {
+			this.tabs.splice(index, 0, tab);
+		} else {
+			this.tabs.push(tab);
+		}
+	}
+
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {string} label
+	 * @param {boolean} disabled
+	 * @param {number} index
+	 */
+	addTabByName(label, disabled, index) {
+		if (core.isString(label) && core.isBoolean(disabled)) {
+			let tab = {
+				label: label,
+				disabled: disabled
+			};
+
+			this.addTab(tab, index);
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.tab = this.tabs.findIndex(item => (!item.disabled || !core.isDef(item.disabled)) && !this.disabled);
```

Comment:
By the way, it would be good to have a test case for the default value of `tab` :)